docs/visualizations.md

```diff
@@ -0,0 +1,120 @@
+# Visualization workflow
+
+## Purpose of visualizations
+Visualizations have two main purposes:
+
+- Display results of computations. Each node can be assigned with a visualization (displayed next to
+  the node). After a node computes its new value, visualization shows it in an understandable way to
+  the user.
+
+- Control the computations. Think about visualizations as a separate component, just as nodes, with 
+  their own input and output ports. Whenever new data appears on the input port, the visualization 
+  displays it. However, when the input port is not connected, the visualization behaves like a 
+  data-producing widget. For example, a map visualization can display locations encoded in its input
+  data. It can also allow to interactively pick new locations with mouse when no input is connected
+  and send the manual selected locations to the output port. In similar fashion histogram can be 
+  used to manually draw histogram with mouse to set data on the next nodes. A number visualization 
+  is at the same time a slider. Image visualization can behave like an image editor, etc.
+
+
+## Visualization Display Forms
+Visualizations can be displayed in the following ways:
+
+- **Attached to nodes**  
+  Their input and output ports are not visible in this form. They are used only to display the 
+  result of the node they are attached to. Whenever you move the node, the visualization moves as 
+  well. The display of visualization in this form can be triggered on selected nodes by clicking the 
+  "space" button. 
+
+- **Fullscreen**  
+  Similar to the previous mode, but the visualization occupies the full visual space. This form can 
```

Comment:
Do you mean here the full IDE size or full screen?

Reply:
full IDE size of course.